app/tests/unit/containers/Admin/ReportingYear/reportingYearValidation.test.ts

```diff
@@ -0,0 +1,582 @@
+    const resultingErrors = validateExclusiveDateRanges(
+      2010,
+      existingYears,
+      {
+        applicationOpenTime: '2011-07-03T00:00:00-07:00',
+        applicationCloseTime: '2011-08-31T23:59:59.999999-07:00',
+        reportingPeriodStart: '2010-01-01T00:00:00-08:00',
+        reportingPeriodEnd: '2010-12-31T23:59:59-08:00'
+      },
+      jsonSchemaErrors
+    );
+    expect(resultingErrors.__errors).not.toContain(
+      ERRORS.APPLICATION_WINDOW_OVERLAPS
+    );
+  });
+  it('Reporting period overlap with that of another existing year should add an error', () => {
+    const resultingErrors = validateExclusiveDateRanges(
+      2030,
+      existingYears,
+      {
+        applicationOpenTime: '2031-07-03T00:00:00-07:00',
+        applicationCloseTime: '2031-08-31T23:59:59.999999-07:00',
+        reportingPeriodStart: '2020-01-01T00:00:00-08:00',
+        reportingPeriodEnd: '2020-12-31T23:59:59-08:00'
+      },
+      jsonSchemaErrors
+    );
+    expect(resultingErrors.__errors).toContain(
+      ERRORS.REPORTING_PERIOD_OVERLAPS
+    );
+  });
+  it('Reporting period that does not overlap with existing years should not add an error', () => {
+    const resultingErrors = validateExclusiveDateRanges(
+      2030,
+      existingYears,
+      {
+        applicationOpenTime: '2031-07-03T00:00:00-07:00',
+        applicationCloseTime: '2031-08-31T23:59:59.999999-07:00',
+        reportingPeriodStart: '2030-01-01T00:00:00-08:00',
+        reportingPeriodEnd: '2030-12-31T23:59:59-08:00'
+      },
+      jsonSchemaErrors
+    );
+    expect(resultingErrors.__errors).not.toContain(
+      ERRORS.REPORTING_PERIOD_OVERLAPS
+    );
+  });
+});
+
+describe('Reporting period admin validation functions: application open/close dates)', () => {
+  let jsonSchemaErrors;
+
+  beforeEach(() => {
+    jsonSchemaErrors = new FakeJsonSchemaErrors();
+  });
+
+  it('Past application open date should add an error', () => {
+    const resultingErrors = validateApplicationDates(
+      existingYears,
+      {
+        reportingPeriodStart: '2010-01-01T00:00:00-08:00',
+        reportingPeriodEnd: '2010-12-31T23:59:59-08:00',
+        applicationOpenTime: '2011-07-03T00:00:00-07:00',
+        applicationCloseTime: '2011-08-31T23:59:59.999999-07:00'
+      },
+      jsonSchemaErrors,
+      {
+        applicationOpenTime: {
+          'ui:disabled': false
+        },
+        applicationCloseTime: {
+          'ui:disabled': false
+        }
+      },
+      true
+    );
+    expect(resultingErrors.__errors).toContain(
+      `${ERRORS.PAST_DATE} Application open time`
+    );
+  });
+  it('Present or future application open date should not add an error', () => {
+    const resultingErrors1 = validateApplicationDates(
+      existingYears,
+      {
+        reportingPeriodStart: '2030-01-01T00:00:00-08:00',
+        reportingPeriodEnd: '2030-12-31T23:59:59-08:00',
+        applicationOpenTime: TODAY,
```

Comment:
I suppose this is fine, as long as we look into mocking the current date before the summer of 2031 :) I really wouldn't bother mocking things now anyway if we're planning on migrating away from Moment.js for our dates